packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -492,21 +492,24 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     }
 
                     // Populate the cache with paths from id-to-path mapping.
-                    for (const blob of this.blobCache.values()) {
-                        const path = blobsIdToPathMap.get(blob.id);
+                    for (const [blobId, blob] of this.blobCache.entries()) {
+                        const path = blobsIdToPathMap.get(blobId);
                         // If this is the first container that was created for the service, it cannot be
                         // the summarizing container (because the summarizing container is always created
                         // after the main container). In this case, we do not need to do any hashing
                         if (path) {
                             // Schedule the hashes for later, but keep track of the tasks
                             // to ensure they finish before they might be used
-                            const hashP = hashFile(IsoBuffer.from(blob.content, blob.encoding)).then((hash: string) => {
+                            const hashP = hashFile(
+                                blob instanceof ArrayBuffer ?
+                                IsoBuffer.from(blob) :
+                                IsoBuffer.from(blob.content, blob.encoding ?? "utf-8"));
+                            const hashP2 = hashP.then((hash: string) => {
```

Comment:
nit: Feels like we don't need `hashP` anymore? And can name `hashP2` to `hashP`.